### DISCUSSION TITLE
EventListenerTouchAllAtOnce incorrect

### DISCUSSION THREAD

**Discussion by NgVThangBz** [axmolengine/axmol#1356] [Q&A] [answered]
I use `EventListenerTouchAllAtOnce` to listen for layer touch events,
but it seems `touches` always = 1, how do I know if I'm touching the screen with 2 fingers?
![all_touch](https://github.com/axmolengine/axmol/assets/125545428/000cbf66-94dc-4bbe-9e8c-40e4a735d805)
![size](https://github.com/axmolengine/axmol/assets/125545428/eaddb377-b9d5-410f-ab89-9d1e481f39d8)

**Comment by aismann** [comment]
Please have a look on cpp-test. There is a multitouch example

**Reply by NgVThangBz** [reply]
Yes, I looked at it, and I don't see the difference between the two codes

**Comment by aismann** [comment]
And  the example works?
Please have a closer look or a comparing tool

**Reply by NgVThangBz** [reply]
example works, but my code not

**Reply by aismann** [reply]
So your code is not the same. 😉

**Reply by NgVThangBz** [reply]
come on, it's only 2 pieces of code, and there can be no difference here, with 1 touch they work fine, but with muilti touch there's no sign of them working, 
you can check it in ios ?

**Reply by aismann** [reply]
Why not use Debugger and check the size of touches?

**Reply by NgVThangBz** [reply]
im check it, and size =1, you can see my code in post

**Reply by aismann** [reply]
You know the Output of the log is on your Debugger Output Window only?

**Reply by aismann** [reply]
Why not make some breakpoints and check it?

**Reply by aismann** [reply]
And last but not least:
Add your snippet on the working cpp test example

**Reply by aismann** [reply]
I think you should usa a variable and using size() only one time

**Reply by NgVThangBz** [reply]
im try with sample code, and never code run in break point
let check it again in ios
![Screenshot 2023-09-26 at 15 32 56](https://github.com/axmolengine/axmol/assets/125545428/8825d20d-8dd2-4e53-bd4b-a9f3dcb774f9)

**Reply by aismann** [reply]
Please use your code in front of the for loop. Cann you log without an if first?

**Reply by NgVThangBz** [reply]
![Screenshot 2023-09-26 at 15 55 41](https://github.com/axmolengine/axmol/assets/125545428/134232ad-f370-4e89-85ef-c8dec5df4b79)

**Reply by aismann** [reply]
Please use axlog without if

**Reply by NgVThangBz** [reply]
@aismann  What are you trying to prove?

**Reply by aismann** [reply]
Axlog ("%d", s);
Or
touches size()

**Reply by aismann** [reply]
Log the size of touches (not any "blabla" text

**Reply by NgVThangBz** [reply]
Do you see yourself doing something pointless?
Of course they will log 1
![Screenshot 2023-09-26 at 16 11 33](https://github.com/axmolengine/axmol/assets/125545428/598cb9dc-331f-4cee-82cc-353c349e1c5d)

**Reply by NgVThangBz** [reply]
why don't you find someone with a mac and ios and try to test it, rather than trying to prove nonsense?

**Reply by aismann** [reply]
I try to help you. Not more. I dont have an ios system. Im out. Sorry no longer time for you.
Maybe you can add this source snipper and not others PLEASE:
`AXLOG("%d",touches.size());`  (no "if" !!!)

and what is the result?
![image](https://github.com/axmolengine/axmol/assets/8652787/79ef7024-8471-4cc4-8b35-8f586486e267)

`Of course they will log 1`  <<< has your system multitouch display?
If Yes: Maybe a bug on axmol/cocos4.0

**Reply by NgVThangBz** [reply]
You can look at my code again, I log the size of touches ( `unsigned long s = touches.size();`), and it's always equal to 1,
and of course my screen has multitouch, i use ipad th7, and the code i used on cocos2dx 3.17 and it works fine

**Reply by aismann** [reply]
What is the size result on cocos2dx?

**Reply by NgVThangBz** [reply]
2, Of course ( number of touch), 
and affter debug in EAGLView-ios.mm in onTouch ( image bellow)
`touches size` is 1, but `(UIEvent*)event` in `allTouches`  `UITouch* touch` have  size 2 
so please check it again.i think it wrong in ios
![Screenshot 2023-09-26 at 16 52 01](https://github.com/axmolengine/axmol/assets/125545428/4f10978e-6e04-4ccb-b029-256947bfa1df)

**Reply by aismann** [reply]
Can you make an Issue with all useful information?
Or better an PR with the fix?
If done please set the answer point too please.

**Comment by aismann** [comment]
It works on Windows. iOS dont have it

**Comment by antn9x** [comment, marked as answer]
Did you enable multiple touch?
[`[__glView setMultipleTouchEnabled:YES];`](https://stackoverflow.com/questions/21576101/how-to-enable-multitouch-in-cocos2d-x-ios-game-using-c)

**Reply by NgVThangBz** [reply]
it seems this is the cause
